### DISCUSSION TITLE
Adding transform through environment variables

### DISCUSSION THREAD

**Discussion by lejsekt** [dotnet/yarp#1430] [Q&A] [answered]
Hi,
I'd like to set RequestHeaderOriginalHost to true for some routes. Is it possible to do it through environment variables? I've tried

```
ReverseProxy__Routes__myRoute__Transforms__0__RequestHeaderOriginalHost: "true"
```

but it doesn't work.

Thank you

**Comment by Tratcher** [comment, marked as answer]
That example worked when I tested it by adding this in Program.Main:
`Environment.SetEnvironmentVariable("ReverseProxy__Routes__route1__Transforms__0__RequestHeaderOriginalHost", "true");`

It combine with:
```
  "ReverseProxy": {
    "Routes": {
      "route1": {
        "ClusterId": "cluster1",
        "Match": {
          "Methods": [ "GET", "POST" ],
          "Hosts": [ "localhost" ],
          "Path": "/api/{**catch-all}"
        }
      },
```

What does the rest of your config look like? Arrays in config can be tricky, I don't think it will work if you already have a transforms section for that route, they'll overlap.

**Reply by Tratcher** [reply]
Why do you need to set just RequestHeaderOriginalHost from environment variables?

Some alternatives:
AddTransforms can be used to programmatically add transforms after loading:
https://github.com/microsoft/reverse-proxy/blob/ec6c5a2be3c5e4dee8b3e24c614dc08bf550f815/samples/ReverseProxy.Transforms.Sample/Startup.cs#L39
IProxyConfigFilter's can be used to modify routes while loading the config.
https://github.com/microsoft/reverse-proxy/blob/ec6c5a2be3c5e4dee8b3e24c614dc08bf550f815/samples/ReverseProxy.ConfigFilter.Sample/CustomConfigFilter.cs#L55

**Reply by samsp-msft** [reply]
<warning: shameless sample promotion>

The [Config filter sample](https://github.com/microsoft/reverse-proxy/tree/main/samples/ReverseProxy.ConfigFilter.Sample) shows how to use the filter's ability to modify routes, and pulls the data out of environment vars - just for scenarios like this  - where expressing the path is not very easy.

**Reply by lejsekt** [reply]
Thank you for a very quick response. I went with the AddTransforms option in Startup, but in case I'll later need a more complicated setup, I'll try the config filter.
